### DISCUSSION TITLE
GLB File Size Inflation

### DISCUSSION THREAD

**Discussion by samsloan** [google/model-viewer#4555] [Q&A] [answered]
Having an issue where after uploading a compressed GLB file (3.2MB) from Blender to the Model Viewer Editor, downloading the scene without making any changes results in a file size increase to 50MB. Attempting to resolve this by re-importing the inflated GLB file back into Blender and re-exporting with compression successfully reduces the size back to around 3MB. 

However, this workaround leads to another issue where the variant names on my model (created with the Model Viewer Editor) are jumbled and don't line up anymore for example "Red" actually gives the Blue variant on the model.

Are there any known workflow mistakes that are leading me here? Otherwise any known work-arounds?

The uncompressed model is attached in the zip, when I upload that to the model editor and download the scene without any changes, the downloaded GLB model is inflated to 50MB.

Thanks!
Sam

[squatRack.zip](https://github.com/google/model-viewer/files/13263516/squatRack.zip)

**Comment by elalish** [comment, marked as answer]
Ugh, sounds like a MV editor bug in increasing the file size and a Blender bug in messing up the variant names. PRs welcome on the editor front; not sure when I'll get to looking into that. File a bug on Blender and meanwhile, I find the glTF extension of VSCode to be very useful for correcting small errors like variant names.
